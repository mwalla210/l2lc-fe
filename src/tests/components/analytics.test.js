import React from 'react'
import renderer from 'react-test-renderer'
import Analytics from '../../components/analytics'
jest.mock('react-chartjs-2', () => ({
  Bar: 'Bar',
  Pie: 'Pie',
}))
jest.mock('reactstrap', () => ({
  ButtonGroup: 'ButtonGroup',
  Button: 'Button',
}))
jest.mock('react-spinners', () => ({
  BarLoader: 'BarLoader',
}))

describe('Analytics', () => {
  it ('Renders (non-loading bar with filters)', () => {
    let options = {
      page: {
        analyticsModelList: [
          {
            model: {
              filters: [
                {
                  type: 'type'
                },
                {
                  type: 'type'
                },
              ],
              setFilteredData: jest.fn(),
              timeFilterData: jest.fn(),
              component: 'bar',
              loading: false,
              currentTimeFrame: 'Year',
              jsData: [],
              yLabel: 'yLabel'
            }
          }
        ]
      }
    }
    const component = renderer.create(
      <Analytics {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    let filterSet = component.root.children[0].children[0].children[1].children[0].children[0].children
    filterSet.forEach(buttonInstance => {
      buttonInstance.props.onClick()
    })
    let timeFrameSet = component.root.children[0].children[0].children[1].children[1].children[0].children
    timeFrameSet.forEach(buttonInstance => {
      buttonInstance.props.onClick()
    })
  })
  it ('Renders (non-loading pie)', () => {
    let options = {
      page: {
        analyticsModelList: [
          {
            model: {
              timeFilterData: jest.fn(),
              component: 'pie',
              loading: false,
              currentTimeFrame: 'Year',
              jsData: [],
              yLabel: 'yLabel'
            }
          }
        ]
      }
    }
    const component = renderer.create(
      <Analytics {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it ('Renders (loading)', () => {
    let options = {
      page: {
        analyticsModelList: [
          {
            model: {
              timeFilterData: jest.fn(),
              component: 'bar',
              loading: true,
              currentTimeFrame: 'Year',
              jsData: [],
              yLabel: 'yLabel'
            }
          }
        ]
      }
    }
    const component = renderer.create(
      <Analytics {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
