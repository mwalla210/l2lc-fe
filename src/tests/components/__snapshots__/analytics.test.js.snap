// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analytics Renders (loading) 1`] = `
<div
  className="row justify-content-center"
>
  <div
    className="col-12"
    style={
      Object {
        "display": "inline-block",
        "paddingTop": 10,
      }
    }
  >
    <h4
      style={
        Object {
          "textAlign": "center",
        }
      }
    />
    <div
      className="row justify-content-center"
      style={
        Object {
          "paddingBottom": "30px",
          "paddingTop": "30px",
        }
      }
    >
      <BarLoader
        color="#2baae2"
        height={8}
        loading={true}
        width={200}
      />
    </div>
  </div>
</div>
`;

exports[`Analytics Renders (non-loading bar with filters) 1`] = `
<div
  className="row justify-content-center"
>
  <div
    className="col-12"
    style={
      Object {
        "display": "inline-block",
        "paddingTop": 10,
      }
    }
  >
    <h4
      style={
        Object {
          "textAlign": "center",
        }
      }
    />
    <div>
      <div
        className="row justify-content-center"
      >
        <ButtonGroup>
          <Button
            active={false}
            color="primary"
            onClick={[Function]}
            size="sm"
          >
            type
          </Button>
          <Button
            active={false}
            color="primary"
            onClick={[Function]}
            size="sm"
          >
            type
          </Button>
        </ButtonGroup>
      </div>
      <div
        className="row justify-content-center"
      >
        <ButtonGroup>
          <Button
            active={true}
            color="secondary"
            onClick={[Function]}
            size="sm"
          >
            Year
          </Button>
          <Button
            active={false}
            color="secondary"
            onClick={[Function]}
            size="sm"
          >
            3 Months
          </Button>
          <Button
            active={false}
            color="secondary"
            onClick={[Function]}
            size="sm"
          >
            1 Month
          </Button>
          <Button
            active={false}
            color="secondary"
            onClick={[Function]}
            size="sm"
          >
            1 Week
          </Button>
        </ButtonGroup>
      </div>
      <Bar
        data={Array []}
        height={60}
        options={
          Object {
            "legend": Object {
              "display": false,
            },
            "responsive": true,
            "scales": Object {
              "xAxes": Array [
                Object {
                  "display": true,
                  "ticks": Object {
                    "autoSkip": false,
                  },
                },
              ],
              "yAxes": Array [
                Object {
                  "display": true,
                  "scaleLabel": Object {
                    "display": true,
                    "labelString": "yLabel",
                  },
                  "ticks": Object {
                    "beginAtZero": true,
                  },
                },
              ],
            },
          }
        }
        width={100}
      />
    </div>
  </div>
</div>
`;

exports[`Analytics Renders (non-loading pie) 1`] = `
<div
  className="row justify-content-center"
>
  <div
    className="col-6"
    style={
      Object {
        "display": "inline-block",
        "paddingTop": 10,
      }
    }
  >
    <h4
      style={
        Object {
          "textAlign": "center",
        }
      }
    />
    <div>
      <div
        className="row justify-content-center"
      >
        <ButtonGroup>
          <Button
            active={true}
            color="secondary"
            onClick={[Function]}
            size="sm"
          >
            Year
          </Button>
          <Button
            active={false}
            color="secondary"
            onClick={[Function]}
            size="sm"
          >
            3 Months
          </Button>
          <Button
            active={false}
            color="secondary"
            onClick={[Function]}
            size="sm"
          >
            1 Month
          </Button>
          <Button
            active={false}
            color="secondary"
            onClick={[Function]}
            size="sm"
          >
            1 Week
          </Button>
        </ButtonGroup>
      </div>
      <Pie
        data={Array []}
        options={
          Object {
            "legend": Object {
              "display": false,
            },
          }
        }
      />
    </div>
  </div>
</div>
`;
